src/core/transaction.c

```diff
@@ -628,7 +628,12 @@ static int transaction_apply(
                 HASHMAP_FOREACH(j, m->jobs) {
                         assert(j->installed);
 
-                        if (j->unit->ignore_on_isolate)
+                        if (j->unit->ignore_on_isolate ||
+                            (IN_SET(manager_state(j->unit->manager),
+                                    MANAGER_INITIALIZING,
+                                    MANAGER_STARTING,
+                                    MANAGER_STOPPING) &&
+                            j->unit->survive))
```

Comment:
the isolate on start-up will stop the unit otherwise

Reply:
this looks like a very special hack. it appears to me, services of this type should really set IgnoreOnIsolate on their own.

Reply:
It's not a hack, it's required to have a service survive a soft reboot, otherwise it gets stopped, which means `SurviveSystemTransition=yes` would not work - the system transition would stop the service